.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,44 @@
+name: Rust Docs to Github Pages
+run-name: ${{ github.actor }} is building and deploying rustdoc to GitHub Pages from ${{ github.ref }}
+# Because documentation versioning and build promotion are not implemente yet, leaving this to be triggered manually only for now.
+on: [workflow_dispatch]
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build-and-deploy-docs:
+    runs-on: ubuntu-22.04
+    # Specifying these flags here as environment variables, because specifying them in the .cargo/config file interferes with the other
+    # workflows, because of the nightly features.
+    env:
+        RUSTDOCFLAGS: "-Zunstable-options --enable-index-page -Aunknown_lints"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Configure GitHub Pages
+      uses: actions/configure-pages@v5
+
+    - name: Build Rust Docs
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly-2023-06-01
```

Comment:
use a newer one, we are no longer locked to old version, since we updated rust version.